server/middleware/errorHandling.ts

```diff
@@ -1,63 +1,43 @@
 import type { MiddlewareError } from '@/types/errorHandling';
+import { CustomError } from '@/utils/customErrors';
 import type { Request, Response } from 'express';
-
-/**
- * Create an error object to be used in the error handling middleware.
- *
- * @param method The method where the error occurred
- * @param controller The controller where the error occurred
- * @param message The error message
- * @param error The error object
- * @param status The status code to return
- *
- * @return the error object
- */
-export function createError(
-  method: string,
-  controller: string,
-  errorString: string,
-  error: unknown,
-): Error {
-  switch (typeof error) {
-    case 'string':
-      errorString = error;
-      break;
-    case 'object':
-      errorString = JSON.stringify(error, Object.getOwnPropertyNames(error));
-      break;
-    default:
-    // TODO: implement logging service
-    // console.error('Error type not handled in createError.');
-    // console.error('Error type:', typeof error);
-  }
-
-  const errorMessage = `Error occured in ${controller}.${method}: ${errorString}`;
-  const wrappedError = new Error(errorMessage);
-
-  return wrappedError;
-}
+import type { NextFunction } from 'express';
+import Log4js from 'log4js';
+const logger = Log4js.getLogger('[ErrorHandler]');
 
 /**
  * Global error handler for the Express server.
  *
  * @param err The error object generated by the middleware
  * @param _ The request object, not used
  * @param res The response object
+ * @param next The next object, not used
  * @return The response object, with error status and error message
  */
+
 export function globalErrorHandler(
   err: Error,
   _: Request,
   res: Response,
+  _next: NextFunction,
 ): Response {
   const defaultError: MiddlewareError = {
     name: 'MiddlewareError',
     log: 'Express error handler caught unknown middleware error',
     status: 500,
     message: { err: 'An error occurred' },
   };
+  const errObj =
```

Comment:
I think the error message structure can be simplified. 
![Pasted Graphic](https://github.com/user-attachments/assets/6a1c3518-21b9-4ef7-97ff-3482fa7b4af3)

Instead of returning an object `message: { err: err.message }`, consider returning a string directly, it will also makes it easier for the frontend to retrieve the error message. The updated code could look like this:
```
export interface MiddlewareError extends Error {
  log: string;
  status: number;
  message: string;
}

  export function globalErrorHandler(
  err: Error,
  _: Request,
  res: Response,
  _next: NextFunction,
): Response {
  const defaultError: MiddlewareError = {
    name: 'MiddlewareError',
    log: 'Express error handler caught unknown middleware error',
    status: 500,
    message: 'An error occurred',
  };
  const errObj =
    err instanceof CustomError
      ? {
          ...defaultError,
          status: err.status,
          message: err.message,
          log: err.log,
        }
      : defaultError;

  logger.error('Error Details:', errObj);

  return res.status(errObj.status).json(errObj.message);
}
```
![Pasted Graphic 2](https://github.com/user-attachments/assets/65221071-e5ee-4e16-9f47-f80e585490e0)

Reply:
refactored to return err string directly now